added scripts to manage user accounts in bulk

d78ui98 opened the pull request:
- [x] added `user-accounts.txt`
- [x] added `accounts.sh`

closes #79

d78ui98 committed:
added accountss.sh

this script it for user management in bulk. it can be used to import, export, backup, restore

d78ui98 committed:
added user-accounts.txt

d78ui98 commented:
the encrypted password becomes the password for the user if we put it in `user-accounts.txt`.

d78ui98 commented:
This code is at its initial stage.
I am working on improving this.

akash0x53 commented on user-accounts.txt:
```diff
@@ -0,0 +1,2 @@
+user1:pass1
```
This can be done by storing as username:<encrypted password> and then use `-p` flag in `useradd` command.

d78ui98 replied:
It is a good idea. 
But only in case of restore. As the password is encrypted.

akash0x53 commented on accounts.sh:
```diff
@@ -0,0 +1,76 @@
+{
+    for line in $(cat $filename)
+        do
+            user=$(echo $line | cut -f1 -d:)
```
Check how you can split string with the help of  IFS. e.g
```
akash@skyfall:~$ a="Akash:Password"
akash@skyfall:~$ echo $a
Akash:Password
akash@skyfall:~$ IFS=":"
akash@skyfall:~$ read -ra ARRAY <<<"$a"
akash@skyfall:~$ echo ${ARRAY[0]}
Akash
akash@skyfall:~$ echo ${ARRAY[1]}
Password
```

d78ui98 replied:
okay. I will try it out.

d78ui98 committed:
using IFS to split string

d78ui98 committed:
added help option

d78ui98 committed:
removed create_user function

d78ui98 committed:
updated import case

d78ui98 committed:
updated export

export case now echos the encrypted passwords to user-accounts.txt and correctly greps the normal user excluding vagrant, vbox and ubuntu

d78ui98 committed:
updated backup case

backup now provides the option to add the archive file to pass in commandline argument, greps the normal users excluding vagrant, vbox etc

d78ui98 committed:
updated restore case

restore now first extracts home directory, user-accounts.txt. then add users with the help of encrypted passwords and later copies important files shel folder

d78ui98 committed:
updated examples, options

akash0x53 commented on accounts.sh:
```diff
@@ -0,0 +1,97 @@
+   
+    export )
+        echo "" > /vagrant/$filename
+        normal_user=$(awk -F: '($3>=1001)&&($1!="nobody"){print $1; exit}' /etc/passwd)
```
it should be >=1000

akash0x53 commented on accounts.sh:
```diff
@@ -0,0 +1,97 @@
+    
+    backup )
+        echo "" > /vagrant/*.txt
+        normal_user=$(awk -F: '($3>=1001)&&($1!="nobody"){print $1; exit}' /etc/passwd)
```
same line here - lets convert this into function.

d78ui98 replied:
other good option is to define this variable with other variables.

akash0x53 commented on accounts.sh:
```diff
@@ -0,0 +1,97 @@
+
+if [[ "$1" == "-h" ]] || [[ "$1" == "--help" ]]  ; then
+   help
+   exit 1
```
exit 0; -h prints the help and generated no error.

akash0x53 commented on accounts.sh:
```diff
@@ -60,20 +60,26 @@ case $1 in
-       ;;
+        echo "" > /vagrant/*.txt
+        normal_user=$(awk -F: '($3>=1001)&&($1!="nobody"){print $1; exit}' /etc/passwd)
+        for user in $(cat /etc/shadow | grep -A 5000 $normal_user | cut -f1-2 -d:)
```
this give me wrong result. I would prefer awk.

d78ui98 replied:
I refined `cut` and `grep`. Now it will work in every case.
 
`awk` creates some problems. It used {} and () brackets. So I am unable to assign output to variable. I could use backticks but it is not a good practice.
So I removed `awk` completely.

d78ui98 committed:
small fixes

d78ui98 committed:
removed redundent lines

d78ui98 committed:
removed awk, refined cut and grep

dashohoxha commented:
I just noticed that PiNet does it as well: http://pinet.org.uk/articles/manage-users/migration.html
You could have checked their code to get some ideas about it. Or maybe you did it, I don't know.
